File: rejabs-backend/src/main/java/com/rejabsbackend/model/Card.java
package com.rejabsbackend.model;

import com.rejabsbackend.enums.Label;
import org.springframework.data.mongodb.core.mapping.Document;

import org.springframework.data.annotation.Id;

import java.time.Instant;
import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Document(collection = "cards")
public record Card(
        @Id
        String cardId,
        String cardTitle,
        String description,
        String listId,
        int position,
        List<Label> labels,
        LocalDate dueDate,
        Instant createdAt,
        Instant updatedAt
) {
}

File: rejabs-frontend/src/types/Card.ts
export type Card = {
    cardId: string,
    cardTitle: string,
    description: string,
    listId: string,
    position: number,
    labels: string[],
    dueDate: string,
    createdAt: Date,
    updatedAt: Date
}

export type CardContextType = {
    cards: Card[],
    setCards: (cards: Card[]) => void
    getCard: (id: string) => Promise<Card>,
    createCard: (data: Partial<Card>) => Promise<Card | null>,
    updateCard: (id: string, data: Partial<Card>) => Promise<Card | null>,
    deleteCard: (id: string) => Promise<boolean>,
    refreshCards: () => Promise<void>,
    error: string,
    setError: (error: string) => void
}
